diff --git a/node_modules/react-native-elements/src/index.d.ts b/node_modules/react-native-elements/src/index.d.ts
index ff9ff85..8aa4354 100644
--- a/node_modules/react-native-elements/src/index.d.ts
+++ b/node_modules/react-native-elements/src/index.d.ts
@@ -830,7 +830,7 @@ export interface InputProps extends TextInputProperties {
   /**
    * Renders component in place of the React Native `TextInput` (optional)
    */
-  inputComponent?: React.ComponentClass<any>;
+  inputComponent?: React.ComponentType<any>;
 
   /**
    * 	Adds styling to input component (optional)
